backend/src/model/game.rs

```diff
@@ -0,0 +1,61 @@
+use crate::db::schema::games;
+use crate::db::schema::games::dsl::*;
+use crate::db::util::{get_conn, DbPool};
+use diesel::{prelude::*, result::Error, Identifiable, Insertable, QueryDsl, Queryable};
+use diesel_async::RunQueryDsl;
+use serde::{Deserialize, Serialize};
+
+#[derive(Insertable, Debug)]
+#[diesel(table_name = games)]
+// TODO use our types
+pub struct NewGame {
+    pub black_uid: String, // uid of user
+    pub game_status: String,
+    pub game_type: String,
+    pub history: String,
+    pub tournament_queen_rule: bool,
+    pub turn: i32,
+    pub white_uid: String, // uid of user
+}
+
+#[derive(Queryable, Identifiable, Serialize, Deserialize, Debug, AsChangeset)]
+#[diesel(primary_key(id))]
+#[diesel(table_name = games)]
+pub struct Game {
+    pub id: i32,
+    pub black_uid: String, // uid of user
+    pub game_status: String,
+    pub game_type: String,
+    pub history: String, //"piece pos;piece pos;piece pos;"
+    pub tournament_queen_rule: bool,
+    pub turn: i32,
+    pub white_uid: String, // uid of user
+}
+
+impl Game {
+    pub async fn create(new_game: &NewGame, pool: &DbPool) -> Result<Game, Error> {
+        let conn = &mut get_conn(pool).await?;
+        new_game.insert_into(games::table).get_result(conn).await
+    }
+
+    pub async fn make_move(&self, board_move: String, pool: &DbPool) -> Result<Game, Error> {
```

Comment:
i think maybe a more ergonomic function would be

```suggestion
    pub async fn update(&mut self, state: &State, pool: &DbPool) -> Result<Game, Error> {
```

since we're going to want to update more than just moves, but also the turn counter and game status from w/e the current state is. if we derive diesel's `AsChangeset`, then we can update `self`'s fields directly and just do

```
diesel::update(games::table).set(self).get_result(conn).await
```